### PR TITLE
Votes consistent permissions

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Feature
- Bugfix

### Detail
The Votes submodule is used by Dashboards, Datasets and Redshift_Datasets in the frontend. Although in Dashboards anyone can see and click "upvote" and in the other 2 modules it is restricted to admins of the resource only.

In this PR we consolidate one single behavior for Votes:
- The UpVote button is visible by all users: any user can get the number of upvotes for a data asset
- The UpVote button is visible but disabled for any user except for the resource admin team: only users with access to the resource can vote for it.

In this PR we mimic these 2 rules also in the backend:
- the upVote API is restricted to users with `GET_X` permissions to the resource
- the getVote and countVotes APIs are open to all users

### Testing
- [X] Deployed to AWS
- [X] Redshift Dataset owner can upvote a dataset
- [X] Programmatically a non-owner receives an error message when executing upvote

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by petrkalos — APPROVED** [review]

File: frontend/src/modules/S3_Datasets/views/DatasetView.js [hunk 1]
```diff
@@ -266,13 +266,14 @@ const DatasetView = () => {
 
             <Grid item>
               <Box sx={{ m: -1 }}>
+                <UpVoteButton
+                  disabled={!isAdmin}
+                  upVoted={isUpVoted}
+                  onClick={() => upVoteDataset(dataset.datasetUri)}
+                  upVotes={upVotes}
+                />
```
Comment (noah-paige) [inline frontend/src/modules/S3_Datasets/views/DatasetView.js:274]: does `UpVote` take in a `disabled` prop / does this really do anything?

Comment (dlpzx) [reply frontend/src/modules/S3_Datasets/views/DatasetView.js:274]: It just shows the number of votes, it allows users to read the upVotes but not to upvote

Comment (noah-paige) [reply frontend/src/modules/S3_Datasets/views/DatasetView.js:274]: Pretty sure I was a user who had share access to the dataset and was still able to execute upVote - only being stopped on the server side

Comment (dlpzx) [reply frontend/src/modules/S3_Datasets/views/DatasetView.js:274]: Fixed, thank you!
image

**Review by noah-paige — APPROVED** [review]

**Pull request merged**